doc/manual/src/architecture/architecture.md

```diff
@@ -0,0 +1,42 @@
+# Architecture
+
+This chapter describes how Nix works.
+It should help users understand why Nix behaves as it does, and it should help developers understand how to modify Nix and how to write similar tools.
+
+## Overview
+
+Nix consists of hierarchical [layers](https://en.m.wikipedia.org/wiki/Multitier_architecture#Layers).
+
+```
+               [ commmand line interface ]
+                            |
+                            | evaluates
```

Comment:
Invokes everything below? Or something to match your

> translating from invocations of Nix executables to interactions with the underlying layers.

better?

(Of course, I am a very much a fan of how one can skip using the expression language :).)

Reply:
Yeah, there is a bit of a mismatch here, and in any case there should be multiple, less graphically crude, and conceptually increasingly refined diagrams showing the interactions. This one illustrates part of `nix-build` to make clear the hierarchical structure. For `nix-store` the CLI talks directly to the store. Editing Nix expressions in the file system is another user interface, in a sense.